### PR TITLE
DADA2 memory optimizations and bugfixes

### PR DESCRIPTION
**Pull request by m-murphy**
Functionally everything is the same, this commit reduces peak memory usage and improves readability. Outputs should not change.
- No longer preload dereplicated files, reducing peak memory usage. DADA2 automatically dereplicates files as needed
- Fix function calls to respect args$cores input
- remove large objects from memory as they are finished being used, reducing peak memory usage especially at end of script
- rewrite the unique allele merging function to be more readable

### CODE REVIEW

**Review by kathryn1995 — APPROVED** [review]
I would wait for Brian's review before merging as he's more familiar with the pipeline, but I think all of these changes look good

**Comment by bgpalmer** [comment]
this looks good to me

**Review by bgpalmer — APPROVED** [review]

**Pull request merged**
